File: src/fr/gstraymond/magicsearch/model/request/facet/Facet.java
package fr.gstraymond.magicsearch.model.request.facet;

public class Facet {

	private Term terms;

	public Facet(String facet) {
		this.terms = new Term(facet);
	}

	public Term getTerms() {
		return terms;
	}

	public void setTerms(Term terms) {
		this.terms = terms;
	}

}

File: app/src/main/kotlin/fr/gstraymond/models/search/request/facet/Facet.kt
package fr.gstraymond.models.search.request.facet

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Facet(val terms: Term) {
    companion object {
        fun fromField(field: String) = Facet(Term(field))
    }

    override fun toString() = "Facet[$terms}"
}

@JsonClass(generateAdapter = true)
data class Term(val field: String, var size: Int? = null) {
    override fun toString() = "Term$field;${size ?: "NA"}"
}